lib/mongodb.js

```diff
@@ -2015,6 +2026,64 @@ MongoDB.prototype.ping = function(cb) {
   }
 };
 
+MongoDB.prototype.beginTransaction = function(isolationLevel, cb) {
+  // TODO: think about how to convert READ_COMMITED, etc. to transactionOptions
+  const transactionOptions = {
+    readPreference: 'primary',
+    readConcern: {level: 'local'},
+    writeConcern: {w: 'majority'},
+  };
+  if (typeof isolationLevel === 'object') {
+    Object.assign(transactionOptions, isolationLevel || {});
+  }
+  const session = this.client.startSession();
+  session.startTransaction(transactionOptions);
+  cb(null, session);
+};
+
+MongoDB.prototype.commit = function(tx, cb) {
+  tx.commitTransaction(function(err) {
+    if (err) {
+      // end session in any case
+      tx.endSession(null, function(error) {
+        if (err) return cb(err);
+        if (error) return cb(error);
+        cb();
+      });
+    } else {
+      tx.endSession(null, function(error) {
+        if (error) return cb(error);
+        cb();
+      });
+    }
+  });
+};
+
+MongoDB.prototype.rollback = function(tx, cb) {
+  tx.abortTransaction(function(err) {
+    if (err) {
+      // end session in any case
+      tx.endSession(null, function(error) {
+        if (err) return cb(err);
+        if (error) return cb(error);
+        cb();
+      });
+    } else {
+      tx.endSession(null, function(error) {
```

Comment:
You duplicate this code. Not so sure if it will work, but I suggest you remove the first ```if (err) {``` because you are ending the session anyway. Also remove the inner ```else {``` to simplify your code.

Reply:
```
MongoDB.prototype.rollback = function(tx, cb) {
  tx.abortTransaction(function(err) {
    tx.endSession(null, function(error) {
      if (err) return cb(err);
      if (error) return cb(error);
      cb();
    });
  });
};
```

like this?

Reply:
Correct!, same for ```MongoDB.prototype.commit```

Reply:
done